home/tests/test_views.py

```diff
@@ -219,32 +204,33 @@ def create_content_page(self):
         Helper function to create pages needed for each test.
         """
         home_page = HomePage.objects.first()
-        main_menu = home_page.get_children().filter(slug="main-menu").first()
-        if not main_menu:
-            main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
-        parent = main_menu
-
-        bodies = []
-
-        bodies.append(WABody("default page", [WABlk("default body")]))
-
+        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
         content_page = PageBuilder.build_cp(
-            parent=parent, slug="default-page", title="default page", bodies=bodies
+            parent=main_menu, slug="default-page", title="default page", bodies=[]
         )
         return content_page
 
-    def test_response_success(self, admin_client):
+    def test_can_edit_page_that_exists(self, admin_client):
         """
-        Check that the edit page can be accessed on a page that exists
+        The edit page can be accessed on a page that exists
         """
         page = self.create_content_page()
         response = admin_client.get(f"/admin/pages/{page.id}/edit/")
 
         assert response.status_code == status.HTTP_200_OK
 
+    def test_page_cannot_be_edited(self, admin_client):
+        """
+        The edit page cannot be accessed on a page that does not exist
+        """
+        response = admin_client.get("/admin/pages/10000/edit/")
+
+        # TODO: a page ID that doesn't exist returns a 302, should this not return a 404?
+        assert response.status_code != status.HTTP_200_OK
```

Comment:
Did you check the return code after merging in `main`? I expect #306 would have fixed that, but maybe it didn't.